### DISCUSSION TITLE
🔄 Reevaluating Multi-Language Support for Better Stride Documentation ✅

### DISCUSSION THREAD

**Discussion by VaclavElias** [stride3d/stride#1740] [Contributors]
This below is being discussed also here https://discord.com/channels/500285081265635328/500295611581464576. 

Hello, friends. I'd like to solicit your opinions on the multi-language support in our Stride Docs. Currently, the Japanese (JP) documentation isn't working properly. While the left side menu functions correctly, clicking on anything results in content being displayed in English (EN). This problem stems from a caching bug/issue within the **docfx**  tool that generated these pages. It appears to have gone unnoticed, but the issue dates back to version **3.0** of the docs.

I've resolved this issue in the new documentation version, which can be found at this link https://www.vaclavelias.com/stride-docs-next/jp/manual/stride-launcher/index.html. However, maintaining support for multiple languages presents significant challenges. Generating multi-language documentation is complex, particularly when the documentation tool doesn't natively support it, and we must customize the process to suit our needs. Not to mention that documentation heavily relies on human intervention for maintenance and updates, often considered tedious by some.

I recently had a discussion with xen2 about this matter, and we're seriously considering limiting the documentation to English only, at least for now. We'd like to know your thoughts on temporarily removing multi-language support until our community grows larger and includes dedicated and passionate documentation contributors.

While I continue working on the new documentation, I do keep supporting multiple languages. However, I want to emphasize that maintaining a multi-language support website requires long-term contributions. If you're interested in contributing and translating the Stride documents, I would be happy to create a dedicated GitHub wiki page with instructions on how to participate. The new version should make the process much easier. If enough dedicated contributors step forward, we would retain multi-language support.
With that said, is there any Japanese contributor who could check the current state of the Japanese Docs https://www.vaclavelias.com/stride-docs-next/jp/? Without dedicated contributors, we may have to limit our focus to English only, and your input could be essential in this decision.

Thank you 🙂

**Comment by BenjaBobs** [comment]
For reference, google translate's version of the japanese docs: [`https://translate.google.com/translate?sl=en&tl=ja&hl=en&u=https://www.vaclavelias.com/stride-docs-next/en`](https://translate.google.com/translate?sl=en&tl=ja&hl=en&u=https://www.vaclavelias.com/stride-docs-next/en)

**Reply by VaclavElias** [reply]
Thank you. We will investigate if we could at least link to this Google translator in some easy way..

**Comment by NicusorN5** [comment]
I vote for keeping the english documentation.

**Comment by Ethereal77** [comment]
As I said on Discord, I think we should keep, expand, and enhance the current english documentation, and remove (for now) all the other translations. If/when we have a bigger community with people willing to spend the time (which can be significant) to keep the docs up to date and of sufficient quality in other languages, then we can start thinking about integrating other translations properly and/or redesigning the current multi-language support properly.

I extend this also to the multi-language support in the editor. Currently the translations are always behind the english version, the XAML syntax is more complex as a result of the current system, and the build system is also more complex as it depends on `Mono.GetText` and custom build steps. All of this complexity for no big gain (currently) as the translations are way behind current.

I say all of this having contributed a part of the spanish translation myself. Obviously I would want in the future the best spanish translation and docs possible, but for now we should focus our efforts where they can make the most difference.

That's my opinion.

**Reply by Kryptos-FR** [reply]
I think we should keep the other languages but as a read-only archive with a call to contributors (maybe with a warning header or something). Some translated pages might be of some value.m, and removing them altogether seems a bit too harsh.

**Reply by Ethereal77** [reply]
I agree for the most part, but let me explain a bit my concerns:

* For the documentation, maybe we should keep the translated docs for now (maybe as readonly as you say). But wouldn't they cause confusion however if a user reads pretty outdated and incomplete documentation in its own language and never discovers that the good documentation is in English?

  The warning header is a good idea, but I think it would be better in the long run to introduce some kind of auto-translation. A doc page that is translated automatically will probably contain many grammatical and context mistakes, but at least it would be up to date.

* However I was mostly talking in my comment about the localization of the editor / libraries. I believe we should focus our efforts on a robust core even if only in English, and then later reintroduce / rearchitect a better localization system. Maybe this will be even easier when the rewrite in Avalonia allows us to use a more modern framework.

**Reply by VaclavElias** [reply]
By the way, regarding the documentation, we are currently talking only about the Japanese language. It is still available but has not been updated for the last few releases. The most recent Japanese version is 4.2, which has rolled over from previous versions and looks like this: https://doc.stride3d.net/latest/jp/manual/index.html. You can see the version inconsistency in the title.

![image](https://github.com/user-attachments/assets/2671da40-b7af-4b71-84fa-cce119ff93ee)

There has also been interest in Chinese, Russian, and I think Spanish translations. However, if I am correct, it has always been only one individual, which is not helpful in the long term. I mentioned somewhere that a few maintainers need to be available for each language to make it worthwhile to maintain. Additionally, I would like to track, somehow and somewhere, who would be interested so the individuals are aware of each other.

**Reply by Ethereal77** [reply]
If we decide to keep the translated docs, I can contribute to the Spanish documentation.

Btw, if I recall correctly, I think I skimmed through the Japanese docs once a long time ago and found some things in there that were not even in the English docs, or at least they were more thoroughly explained or with more images. Can someone confirm this?

**Reply by VaclavElias** [reply]
> Btw, if I recall correctly, I think I skimmed through the Japanese docs once a long time ago and found some things in there that were not even in the English docs, or at least they were more thoroughly explained or with more images. Can someone confirm this?

I can't but might be true if the JP docs were written when/while JP devs were around from the Silicon Studio from the Paradox/Xenko time.

**Comment by VaclavElias** [comment]
I will be re-evaluating multi-language support **again** and find a way to track how many contributors would like to maintain and help with other languages. That should help us to figure out where to go 🙂.
